mozci/data/contract.py

```diff
@@ -3,18 +3,51 @@
 from dataclasses import dataclass
 from typing import Tuple
 
-from voluptuous import Schema
+from voluptuous import Any, Marker, Optional, Required, Schema
 
 
 @dataclass
 class Contract:
     name: str
+    description: str
     validate_in: Schema
     validate_out: Schema
 
 
-_contracts: Tuple[Contract] = (
-    Contract(name="placeholder", validate_in=Schema({}), validate_out=Schema({}),),
+_contracts: Tuple[Contract, ...] = (
```

Comment:
The only drawback I see is when two datasources underlying DBs are mismatching.
E.g. on AD maybe we can get the push_tasks data with one single query and the push_test_groups with a separate query, on Treeherder maybe it's two queries for push_tasks data and one of the two would also get data for push_test_groups.
So, our contracts will always be optimized better for one datasource rather than another.

Reply:
Yeah, I think we shouldn't hesitate to refactor contracts into smaller ones or change them around if we need to. E.g, we could break this up into pieces, but if using AD still query the parts that aren't needed right away so they're available later.